crates/core/src/config.rs

```diff
@@ -93,8 +122,39 @@ pub struct LogConfig {
     pub directives: Vec<String>,
 }
 
-impl ConfigFile {
-    pub fn read(path: &ConfigToml) -> anyhow::Result<Option<Self>> {
-        parse_config(path.as_ref())
+/// Patch the value of a key in a TOML document,
+/// preserving the formatting and comments of the original value.
+fn patch_value(item: Option<&Item>, value: Option<&str>) -> Option<Item> {
+    match (value, item) {
+        (Some(value), Some(Item::Value(Value::String(v)))) => {
+            let mut new = Value::String(Formatted::new(value.to_string()));
+            let decor = new.decor_mut();
+            *decor = v.decor().clone();
```

Comment:
why do we only do this `decor` dance if the original value is a string? It looks like any `Table` can have `decor`, is that right?